RoboticonColony/Assets/Code/HumanPlayer.cs

```diff
@@ -0,0 +1,431 @@
+﻿using System;
+
+public class HumanPlayer : AbstractPlayer
+{
+    /// <summary>
+    /// Create a HumanPlayer with the given inputController
+    /// </summary>
+    /// <param name="inputController">object to gather user input</param>
+    /// <param name="inv">a player inventory</param>
+    /// <param name="market">the market</param>
+    /// <param name="estateAgent">the estate agent</param>
+    public HumanPlayer(
+        InputController inputController,
+        Inventory inv, 
+        Market market,
+        EstateAgent estateAgent) : base(inputController, inv, market, estateAgent)
+    {
+    }
+
+    /// <summary>
+    /// The player may choose to buy a plot of land
+    /// </summary>
+    /// <param name="timeout">a time for which the phase can run</param>
+    public override void DoPhaseOne(Timeout timeout)
+    {
+        // check the player has enough money for at least the cheapest tile
+        int CheapestTilePrice = int.MaxValue;
+        foreach (Tile UnsoldTile in EstateAgent.GetAvailableTiles())
+        {
+            int CurrentTilePrice = EstateAgent.GetPrice(UnsoldTile);
+            if (CurrentTilePrice < CheapestTilePrice)
+            {
+                CheapestTilePrice = CurrentTilePrice;
+            }
+        }
+        if (Inventory.Money < CheapestTilePrice)
+        {
+            try
+            {
+                Input.Confirm("You dont have enough money to buy a tile this round", timeout);
+            }
+            catch (TimeoutException)
+            {
+                return;
+            }
+            return;
+        }
+
+        // does the player want to buy a tile?
+        bool WishesToPurchase = false;
+        try
+        {
+            WishesToPurchase = Input.PromptBool("Do you wish to purchase a tile?", timeout);
+        }
+        catch (TimeoutException)
+        {
+            return;
+        }
+
+        // loop until timeout, cancel, or tile purchased
+        while (WishesToPurchase)
+        {
```

Comment:
When you call Input, you are just displaying a UI element to the player. The player still has to click the button or do whatever input they need to in their own time. But the code block will never leave the loop to give unity the chance to even register any input. The program still needs to run when you do that, so unity can detect if the mouse is moving over any buttons etc. But the program will never get to that point, as update is called every frame, and you never leave the current frame because you're stuck in a while loop.

Using monobehaviors is a lot different to stuff you've done before, and can seem a bit backwards. Instead of methods like Input.Confirm, which tries to display the input message box and get the input in the same frame, you should have something along the lines of Input.ShowConfirm and Input.ConfirmClicked, then assign Input.ConfirmClicked to the confirm button in the UI (you can attach functions to UI elements for certain events). I know it seems frustrating, but it's just how Unity works.